creator-node/src/components/healthCheck/healthCheckComponentService.js

```diff
@@ -29,7 +33,67 @@ const healthCheck = async ({ libs } = {}, logger, sequelize) => {
   // we have a /db_check route for more granular detail, but the service health check should
   // also check that the db connection is good. having this in the health_check
   // allows us to get auto restarts from liveness probes etc if the db connection is down
-  await sequelize.query('SELECT 1')
+  const [ databaseLiveness ] = await getMonitors([MONITORS.DATABASE_LIVENESS])
```

Comment:
i think we should leave this query as is just for the base health check route, running it on a production content node db takes 0.08 seconds and since we have hundreds of connections in our pool throughput shouldnt be an issue. we've seen our connection manager go bad and if somehow at the same time there's a bug where the cached monitor value is incorrect or isn't getting updated, we may not get auto restarts from kube.

i'm fine having the other routes like ipfs and db_check using the monitor because they don't trigger external actions

Reply:
I actually think this method might be better because we wouldn't take down the service and restart under load / contribute extra load (especially since databases are run co located with other pods on the same machine). Not a strong opinion though, happy to leave as is but I think it is kinda weird that every time a client would hit health check we do a select 1. 

@SidSethi wanted me to replace all the queries here w/ the new monitoring stuff. what're your thoughts?

Reply:
Code here triggering an external action seems kinda like an antipattern too

Reply:
using monitors is definitely the right move for most of these routes, but given the importance of the db i feel like we should have at least one place that queries it for real. we've seen bugs where caches don't get updated as expected like the bug in discovery showing 0 block diff (discovery got hammered by health checks because of service selection so understandable why it had to be done there). not saying there's a bug here, but for the small overhead we pay vs the guarantee i feel like the tradeoff seems worthwhile?

also generally i agree with you about external actions, but i would argue health check is the only exception. kube, aws and a lot of other systems use this as a probe to perform actions.

Reply:
im still not totally convinced, but reverted for now.

Reply:
it seems reasonable to me to have this route use monitor value, bc making 1 db query for every health check call is pretty bad. i agree with you dheeraj that kube, k8s etc should have a way to override. could see this route using monitor by default and k8s / aws pass in a flag to override and query directly

but don't feel strongly about whether or not that should be part of this work item

Reply:
will leave as is for now & revisit later :)